services/openvpn/connection_factory.go

```diff
@@ -75,19 +94,20 @@ func (op *ProcessBasedConnectionFactory) newBytecountMiddleware() management.Mid
 }
 
 func (op *ProcessBasedConnectionFactory) newStateMiddleware(session session.ID, signer identity.Signer, connectionOptions connection.ConnectOptions, stateChannel connection.StateChannel) management.Middleware {
-	// TODO Move stats.Sender from 'openvpn' package
-	statsSender := stats.NewRemoteStatsSender(
+	statsSender := op.statsSenderFactory(
```

Comment:
Why that, statistics sender is leaked into openvpn service, and cannot be simply a part of conn manager? Each connection (a.k.a openvpn or other) should provide clear callbacks - stats, events etc. Conn manager handles them appropriatly